cmd/snap/cmd_set_test.go

```diff
@@ -127,6 +127,97 @@ func (s *snapSetSuite) TestSnapSetFailParsingWithStrictJSON(c *check.C) {
 	c.Assert(err, check.ErrorMatches, "failed to parse JSON:.*")
 }
 
+func (s *snapSetSuite) TestSnapSetFailJsonKeysValidation(c *check.C) {
+	invalidJson := []string{
+		`key={"key0":"val0","key1":"val1","key2":{"key3":"val2","key4":"val3","key5":{"INVALID":"val4","key7":"val5"}},"key8":{"key9":{"key10":"val6"}}}`,
+		`key={"key0":"val0","key1":"val1","key2":{"key3":"val2","key4":"val3",` +
+			`"key5":{"key6":"val4","key7":"val5"}},"key8":{"key9":{"key10":"val6"}},` +
+			`"key11":[{"key12":"val7"},{"KEY13":"val8"},{"key14":["val9","val10"]}]}`,
+	}
+
+	for _, json := range invalidJson {
+		_, err := snapset.Parser(snapset.Client()).ParseArgs([]string{"set", "snapname", "-t", json})
+		c.Assert(err, check.ErrorMatches, "invalid option name:.*")
+	}
+}
+
+func (s *snapSetSuite) TestSnapSetPassJsonKeysValidation(c *check.C) {
```

Comment:
```suggestion
func (s *snapSetSuite) TestValidateJSONKeysHappy(c *check.C) {
```

Reply:
As above, would be great to have a `TestValidateJSONKeysUnhappy` which is just like this but with an invalid key in each case.